filament/src/details/Material.cpp

```diff
@@ -71,6 +74,7 @@ struct Material::BuilderDetails {
     size_t mSize = 0;
     MaterialParser* mMaterialParser = nullptr;
     bool mDefaultMaterial = false;
+    std::unordered_map<const char*, std::variant<int32_t, float, bool>> mConstantSpecializations;
```

Comment:
if we assume "not many" spec constants, we could just use a vector here and do a linear search. Also could we use `std::string_view` instead of `const char*`, this might save some calls to `strlen`.

Reply:
I just realized something. I shouldn't be using a `const char*` here, because the user could release the string after calling `.constant`. We need to persist the string until the material is actually compiled. Likewise I can't use `string_view` either. I should be using `std::string` or `CString` instead.